tests/flask_stub_server.py

```diff
@@ -0,0 +1,150 @@
+
+
+@app.errorhandler(404)
+def error():  # noqa: ANN201
+    data = {"message": "Some response data"}
+    return jsonify(data), 404
+
+
+@app.route("/api/v1/models/create/new", methods=["POST"])
+def model_create():  # noqa: ANN201
+    data = {"status": "success", "message": "Model created successfully"}
+    return jsonify(data), 200
+
+
+@app.route("/api/v1/models/<org_slug>/<model_slug>/create/instance", methods=["POST"])
+def model_instance_create_instance(org_slug: str, model_slug: str):  # noqa: ANN201
+    post_data = request.get_json()
+    # TODO: error here
+    if post_data.get("licenseName", "") not in ALLOWED_LICENSE_VALUES:
+        data = {"error": f"bad: {request.path}"}
+        return jsonify(data), 200
+    data = {"status": "success", "message": f"Model Instance {org_slug}/{model_slug} created successfully"}
+    return jsonify(data), 200
+
+
+@app.route("/api/v1/models/<org_slug>/<model_slug>/<framework>/<variation>/create/version", methods=["POST"])
+def model_instance_create_version(org_slug: str, model_slug: str, framework: str, variation: str):  # noqa: ANN201
+    post_data = request.get_json()
+    # TODO: error here
+    if post_data.get("licenseName", "") not in ALLOWED_LICENSE_VALUES:
+        data = {"error": f"bad: {request.path}"}
+        return jsonify(data), 200
+    data = {
+        "status": "success",
+        "message": f"Model Version {org_slug}/{model_slug}/{framework}/{variation} created successfully",
+    }
+    return jsonify(data), 200
+
+
+@app.route("/api/v1/blobs/upload", methods=["POST"])
+def blob_upload():  # noqa: ANN201
+    if shared_data["simulate_308"] and shared_data["blob_request_count"] < 0:
+        _increment_blob_request()
+        return "", 308
+    data = {
+        "token": "dummy",
+        "createUrl": "http://127.0.0.1:7777//upload/storage/v1/b/kaggle-models-data/o",
+        "status": "success",
+        "message": "Here is your token and Url",
+    }
+    post_data = request.get_json()
+    _add_file(post_data["name"])
+    _increment_blob_request()
+    return jsonify(data), 200
+
+
+@app.route("/upload/storage/v1/b/kaggle-models-data/o", methods=["PUT"])
+def model_instance_create():  # noqa: ANN201
+    data = {"status": "success", "message": "File uploaded"}
+    return jsonify(data), 200
+
+
+class ServerThread(threading.Thread):
+    def __init__(self, app: Flask):
+        threading.Thread.__init__(self)
+        self.server = make_server("127.0.0.1", 7777, app)
+        self.ctx = app.app_context()
+
+    def run(self) -> None:
+        self.server.serve_forever()
+
+    def shutdown(self) -> None:
+        self.server.shutdown()
+
+
+def start_server() -> None:
+    global server  # noqa: PLW0603
+    server = ServerThread(app)
+    server.start()
+
+
+def stop_server() -> None:
+    global server  # noqa: PLW0602
+    server.shutdown()
+
+
+if __name__ == "__main__":
+    app.run("127.0.0.1", port=7777, debug=False, use_reloader=False)
```

Comment:
Let's get the address and port from this environment variable: https://github.com/Kaggle/kagglehub/blob/9c25bbfe2bcfcce847cbb5ddbe080b607f63d153/tests/__init__.py#L7

This is the environment variable that configures the client to point to this test server.

Reply:
Done